azure/scope/managedcontrolplane.go

```diff
@@ -102,13 +105,33 @@ type ManagedControlPlaneScope struct {
 	Client         client.Client
 	patchHelper    *patch.Helper
 	kubeConfigData []byte
+	cache          *ManagedControlPlaneCache
 
 	AzureClients
 	Cluster             *clusterv1.Cluster
 	ControlPlane        *infrav1exp.AzureManagedControlPlane
 	ManagedMachinePools []ManagedMachinePool
 }
 
+// ManagedControlPlaneCache stores common ManagedControlPlane information so we don't have to hit the API multiple times within the same reconcile loop.
+type ManagedControlPlaneCache struct {
```

Comment:
can we do the same thing in cluster scope?

Reply:
done